azext_edge/edge/providers/orchestration/resources/dataflows.py

```diff
@@ -162,29 +184,205 @@ def show(
+                dataflow_name=name,
+                instance_name=instance_name,
+                resource_group_name=resource_group_name,
+            )
+            return wait_for_terminal_state(poller, **kwargs)
+
+    def _validate_dataflow_config(
+        self,
+        dataflow_config: dict,
+        instance_name: str,
+        resource_group_name: str,
+    ):
+        operations = dataflow_config.get("operations", [])
+
+        # get source endpoint
+        source_endpoint_obj = self._process_exist_endpoint(
+            operations=operations,
+            instance_name=instance_name,
+            resource_group_name=resource_group_name,
+            operation_type=DataflowOperationType.SOURCE.value,
+        )
+
+        # validate source endpoint type
+        source_endpoint_type = source_endpoint_obj.get("properties", {}).get("endpointType", "")
+        if source_endpoint_type not in [
+            DataflowEndpointType.KAFKA.value,
+            DataflowEndpointType.MQTT.value,
+        ]:
+            raise InvalidArgumentValueError(
+                f"'{source_endpoint_type}' is not a valid type for source dataflow endpoint."
+            )
+
+        # when Kafka endpoint, validate consumer group id
+        if source_endpoint_type == DataflowEndpointType.KAFKA.value:
+            group_id = source_endpoint_obj.get("properties", {}).get("kafkaSettings", {}).get("consumerGroupId", "")
+            if not group_id:
+                raise InvalidArgumentValueError(
+                    "'consumerGroupId' is required in kafka source dataflow endpoint configuration."
+                )
+
+        # get destination endpoint
+        desination_endpoint_obj = self._process_exist_endpoint(
+            operations=operations,
+            instance_name=instance_name,
+            resource_group_name=resource_group_name,
+            operation_type=DataflowOperationType.DESTINATION.value,
+        )
+
+        trans_operation_type = DataflowOperationType.TRANSFORMATION.value
+        transformation_operation = self._get_operation(
+            operations=operations,
+            operation_type=trans_operation_type,
+        )
+        schema_ref = transformation_operation.get(
+            DATAFLOW_OPERATION_TYPE_SETTINGS[trans_operation_type], {}).get("schemaRef", "")
+
+        # validate schema_ref for destination endpoint type
+        destination_endpoint_type = desination_endpoint_obj.get("properties", {}).get("endpointType", "")
+        if destination_endpoint_type in [
+            DataflowEndpointType.DATAEXPLORER.value,
+            DataflowEndpointType.DATALAKESTORAGE.value,
+            DataflowEndpointType.FABRICONELAKE.value,
+            DataflowEndpointType.LOCALSTORAGE.value,
+        ] and not schema_ref:
+            raise InvalidArgumentValueError(
+                f"'schemaRef' is required for dataflow due to destination endpoint '{destination_endpoint_type}' type."
```

Comment:
The destination endpoint '{destination_endpoint_type}' type requires 'schemaRef'

Reply:
or if you dont feel like changing 290 to be subject first - 

'schemaRef' is required for destination endpoint '{destination_endpoint_type}' type.